napari/_qt/qt_main_window.py

```diff
@@ -578,6 +578,13 @@ def add_dock_widget(
             shortcut=shortcut,
         )
         self._add_viewer_dock_widget(dock_widget)
+
+        if hasattr(widget, 'reset_choices'):
+            # Keep the dropdown menus in the widget in sync with the layer model
+            # if widget has a `reset_choices`, which is true for all magicgui
+            # widgets
+            self.qt_viewer.viewer.layers.events.connect(widget.reset_choices)
```

Comment:
I'm forgetting at the moment... is `layers.events` the same as "any event" or did we need to do `layers.events.inserted` and `layers.events.removed`?

Reply:
`layers.events` will do all of them